src/atoms/Button/Button.web.js

```diff
@@ -1,20 +1,471 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import styled, { useTheme } from 'styled-components';
 
-const Button = ({ type, onClick, children, ...rest }) => (
-  <button onClick={onClick} type={type} {...rest}>
-    {children}
-  </button>
-);
+import Flex from '../Flex';
+import Size from '../Size';
+import Space from '../Space';
+import Icon from '../Icon';
+import Text from '../Text';
+import View from '../View';
+import automation from '../../_helpers/automation-attributes';
+import { getColor, makePxValue, getVariantColorKeys } from '../../_helpers/theme';
+
+const BORDER_WIDTH = makePxValue(0.125);
+
+const relativePadding = (variant, padding, border = BORDER_WIDTH) => {
+  if (variant === 'tertiary') {
+    return makePxValue(padding);
+  }
+  return makePxValue(`${parseFloat(padding) - parseFloat(border)}px`);
+};
+
+const styles = {
+  fontColor({ variant, variantColor, disabled }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return 'light.950';
+        }
+        return 'light.900';
+      case 'secondary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor}.800`;
+      case 'tertiary':
+        if (disabled) {
+          return 'shade.940';
+        }
+        return `${variantColor}.800`;
+      default:
+        if (disabled) {
+          return 'light.950';
+        }
+        return 'light.900';
+    }
+  },
+  backgroundColor({ variant, variantColor, disabled, theme }) {
+    switch (variant) {
+      case 'primary':
+        if (disabled) {
+          return getColor(theme, `${variantColor}.500`);
+        }
+        return getColor(theme, `${variantColor}.800`);
+      case 'secondary':
+        if (disabled) {
+          return getColor(theme, 'shade.910');
+        }
+        return getColor(theme, `${variantColor}.920`);
+      case 'tertiary':
+        return 'transparent';
+      default:
+        if (disabled) {
+          return getColor(theme, `${variantColor}.500`);
+        }
+        return getColor(theme, `${variantColor}.800`);
```

Comment:
should we be explicit about things like this? i.e the `default` case here  is returning the same thing as `primary` and since `primary` is already the `default variant`, do we need to have this `default` case?